dataloader.py

```diff
@@ -0,0 +1,103 @@
+# PyTorch imports
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.sampler import SubsetRandomSampler
+
+# Other libraries for data manipulation and visualization
+import os
+import numpy as np
+
+
+class PadSequence:
+    def __call__(self, batch):
+        #Each element in batch is (features, labels)
+        sorted_batch = sorted(batch, key=lambda x: x[0].shape[0], reverse=True)
+
+        sequences = [x[0] for x in sorted_batch]
+        sequences_padded = torch.nn.utils.rnn.pad_sequence(sequences, batch_first=True)
+
+        labels = [x[1] for x in sorted_batch]
+        labels_padded = torch.nn.utils.rnn.pad_sequence(labels, batch_first=True)
+        return sequences_padded, labels_padded
+
+class PreprocessedParsedReplayDataset(Dataset):
+    """
+    Customed preprocessed input files for parsed game replays.
+    Features/labels should be in folder ./data/feautures and ./data/labels respectively, unless specified.
+    Each txt file in the folder is for 1 game.
+    """
+    
+    def __init__(self, feature_folder='./data/features/', label_folder='./data/labels/'):
+        
+        self.feature_folder = feature_folder
+        self.label_folder = label_folder
+        assert len(os.listdir(self.feature_folder))==len(os.listdir(self.label_folder)), "Number of files in feature and label folders do not match."
+        self.num_games = len(os.listdir(self.feature_folder))
+        
+    def __len__(self):
+        return self.num_games
+    
+    def __getitem__(self, ind):
+        
+        feature_path = os.path.join(self.feature_folder,str(int(ind))+'.txt')
+        label_path = os.path.join(self.label_folder,str(int(ind))+'.txt')
+        
+        features = np.loadtxt(feature_path)
+        labels = np.loadtxt(label_path)
+        
+        return torch.Tensor(features), torch.Tensor(labels)
+
+
+def split_dataloader(batch_size=1, total_num_games=-1, p_val=0.1, p_test=0.2, seed=3154, shuffle=True):
+    
+    dataset = PreprocessedParsedReplayDataset()
```

Comment:
This class has keyword args that you aren't using. If they are allowed, they should be passed in in `split_datdaloader` as well.